File: frontend-web/SAVELYWEB/src/app/donnees/modeles/enums.ts
/** * Fichier regroupant toutes les énumérations du système 
 * Aligné avec le package com.collecte_epargne.collecte_epargne.utils
 */

export enum CategorieNotification {
    TRANSACTION = 'TRANSACTION',
    RAPPEL_COTISATION = 'RAPPEL_COTISATION',
    ALERTE = 'ALERTE',
    PROMO = 'PROMO'
}

export enum FormatRecu {
    PDF = 'PDF',
    SMS = 'SMS',
    EMAIL = 'EMAIL'
}

export enum FrequenceCotisation {
    QUOTIDIEN = 'QUOTIDIEN',
    HEBDOMADAIRE = 'HEBDOMADAIRE',
    MENSUEL = 'MENSUEL'
}

export enum ModeTransaction {
    ONLINE = 'ONLINE',
    OFFLINE = 'OFFLINE'
}

export enum StatutCompte {
    ACTIF = 'ACTIF',
    INACTIF = 'INACTIF',
    BLOQUE = 'BLOQUE',
    CLOTURE = 'CLOTURE'
}

export enum StatutGenerique {
    ACTIF = 'ACTIF',
    INACTIF = 'INACTIF',
    SUSPENDU = 'SUSPENDU'
}

export enum StatutPlanCotisation {
    ACTIF = 'ACTIF',
    TERMINE = 'TERMINE',
    SUSPENDU = 'SUSPENDU'
}

export enum StatutSynchroOffline {
    EN_ATTENTE = 'EN_ATTENTE',
    SYNCHRONISE = 'SYNCHRONISE',
    ERREUR = 'ERREUR',
    SYNCHRONISEE = 'SYNCHRONISEE',
    DOUBLON = 'DOUBLON'
}

export enum StatutTransaction {
    EN_ATTENTE = 'EN_ATTENTE',
    VALIDEE_CAISSE = 'VALIDEE_CAISSE',
    VALIDEE_SUPERVISEUR = 'VALIDEE_SUPERVISEUR',
    TERMINEE = 'TERMINEE',
    ANNULEE = 'ANNULEE',
    REJETEE = 'REJETEE'
}

export enum TypeCNI {
    PASSEPORT = 'PASSEPORT',
    CARTE_IDENTITE = 'CARTE_IDENTITE',
    PERMIS_CONDUIRE = 'PERMIS_CONDUIRE'
}

export enum TypeEmploye {
    SUPERVISEUR = 'SUPERVISEUR',
    CAISSIER = 'CAISSIER',
    COLLECTEUR = 'COLLECTEUR'
}

export enum TypeNotification {
    SMS = 'SMS',
    EMAIL = 'EMAIL',
    PUSH = 'PUSH',
    IN_APP = 'IN_APP'
}

export enum TypeTransaction {
    DEPOT = 'DEPOT',
    RETRAIT = 'RETRAIT',
    REVERSEMENT = 'REVERSEMENT',
    PENALITE = 'PENALITE',
    BONUS = 'BONUS',
    INTERETS = 'INTERETS'
}

File: frontend-mobile/savely/lib/models/enums.dart
/// Enum pour les types de pièce d'identité
enum TypeCni {
  CNI('CNI'),
  PASSPORT('PASSPORT'),
  PERMIS_CONDUIRE('PERMIS_CONDUIRE'),
  AUTRE('AUTRE');

  final String value;
  const TypeCni(this.value);

  factory TypeCni.fromString(String value) {
    return TypeCni.values.firstWhere(
      (e) => e.value == value,
      orElse: () => TypeCni.AUTRE,
    );
  }
}

/// Enum pour les statuts de compte
enum StatutCompte {
  OUVERT('OUVERT'),
  BLOQUE('BLOQUE'),
  CLOTURE('CLOTURE'),
  GELE('GELE'),
  SUSPENDU('SUSPENDU');

  final String value;
  const StatutCompte(this.value);

  factory StatutCompte.fromString(String value) {
    return StatutCompte.values.firstWhere(
      (e) => e.value == value,
      orElse: () => StatutCompte.OUVERT,
    );
  }
}

/// Enum pour le statut d'approbation de compte
enum StatusApprobation {
  EN_ATTENTE('EN_ATTENTE'),
  APPROUVE('APPROUVE'),
  REJETE('REJETE');

  final String value;
  const StatusApprobation(this.value);

  factory StatusApprobation.fromString(String value) {
    return StatusApprobation.values.firstWhere(
      (e) => e.value == value,
      orElse: () => StatusApprobation.EN_ATTENTE,
    );
  }
}

/// Enum pour les types de transaction
enum TypeTransaction {
  DEPOT('DEPOT'),
  RETRAIT('RETRAIT'),
  COTISATION('COTISATION'),
  INTERET('INTERET'),
  PENALITE('PENALITE');

  final String value;
  const TypeTransaction(this.value);

  factory TypeTransaction.fromString(String value) {
    return TypeTransaction.values.firstWhere(
      (e) => e.value == value,
      orElse: () => TypeTransaction.DEPOT,
    );
  }
}

/// Enum pour le statut de transaction
enum StatutTransaction {
  EN_ATTENTE('EN_ATTENTE'),
  VALIDEE_CAISSE('VALIDEE_CAISSE'),
  VALIDEE_SUPERVISEUR('VALIDEE_SUPERVISEUR'),
  TERMINEE('TERMINEE'),
  ANNULEE('ANNULEE'),
  REJETEE('REJETEE');

  final String value;
  const StatutTransaction(this.value);

  factory StatutTransaction.fromString(String value) {
    return StatutTransaction.values.firstWhere(
      (e) => e.value == value,
      orElse: () => StatutTransaction.EN_ATTENTE,
    );
  }
}

/// Enum pour le statut de validation
enum StatusValidation {
  EN_ATTENTE('EN_ATTENTE'),
  VALIDEE('VALIDEE'),
  REJETEE('REJETEE');

  final String value;
  const StatusValidation(this.value);

  factory StatusValidation.fromString(String value) {
    return StatusValidation.values.firstWhere(
      (e) => e.value == value,
      orElse: () => StatusValidation.EN_ATTENTE,
    );
  }
}

/// Enum pour le mode de transaction
enum ModeTransaction {
  LIQUIDE('LIQUIDE'),
  CHEQUE('CHEQUE'),
  VIREMENT('VIREMENT'),
  MOBILE_MONEY('MOBILE_MONEY');

  final String value;
  const ModeTransaction(this.value);

  factory ModeTransaction.fromString(String value) {
    return ModeTransaction.values.firstWhere(
      (e) => e.value == value,
      orElse: () => ModeTransaction.LIQUIDE,
    );
  }
}

/// Enum pour le statut générique (client, utilisateur)
enum StatutGenerique {
  ACTIF('ACTIF'),
  INACTIF('INACTIF'),
  SUSPENDU('SUSPENDU');

  final String value;
  const StatutGenerique(this.value);

  factory StatutGenerique.fromString(String value) {
    return StatutGenerique.values.firstWhere(
      (e) => e.value == value,
      orElse: () => StatutGenerique.ACTIF,
    );
  }
}

/// Enum pour les types d'employé
enum TypeEmploye {
  COLLECTEUR('COLLECTEUR'),
  CAISSIER('CAISSIER'),
  SUPERVISEUR('SUPERVISEUR'),
  ADMIN('ADMIN');

  final String value;
  const TypeEmploye(this.value);

  factory TypeEmploye.fromString(String value) {
    return TypeEmploye.values.firstWhere(
      (e) => e.value == value,
      orElse: () => TypeEmploye.COLLECTEUR,
    );
  }
}
